script.js

```diff
@@ -1,70 +1,96 @@
 (function(global) {
 
-  var requestData = $.getJSON('data.json').done(function(products) {
-    var requestProductTemplate = $.get('product-template.html').done(function(template) {
+  var _template = function(str, param) {
+    return str.replace(/\{(.*?)\}/g, function(match, token) {
+      return param[token];
+    });
+  }
 
-      var data = products;
-      var template = template;
+  var requestData = $.getJSON('data.json').done(function(productData) {
+    var requestProductTemplate = $.get('product-template.html').done(function(productTemplateHtml) {
 
       function TOMApp() {
-        this.products = [];
-      }
-
-      TOMApp.prototype.getproducts = function() {
         var self = this;
-        for (i=0; i<data.sales.length ; i++) {
-          self.products.push(new productobj(data.sales[i], i)  );
+
+        self.regions = {
+          "$mainRegion": $("#content")
         }
+
+        self.products = [];
+
+        self.initProducts();
+        self.render();
       }
 
-      TOMApp.prototype.updateproducthtml = function() {
-        for( i=0; i< this.products.length ; i++){
-          this.products[i].updatehtml();
+      TOMApp.prototype.initProducts = function() {
+        var self = this;
+
+        for (var i=0; i<productData.sales.length ; i++) {
+          self.products.push(new Product(productData.sales[i], i)  );
         }
       }
 
-      TOMApp.prototype.updatedom = function() {
-        thishtml=''; 
-        for(var i=0; i< this.products.length ; i++){
+      TOMApp.prototype.render = function() {
+        var self = this;
+
+        self.regions.$mainRegion.empty();
+
+        var $row = null;
+        for(var i=0; i< self.products.length ; i++){
           
           if (i % 3 == 0 ) {  
-            thishtml += "<div class='row'>"; 
+            $row = $("<div>").addClass("row");
             console.log("START");
           }
 
-          thishtml += this.products[i].htmlview;
-          if ((i % 3 == 2) || i == (this.products.length-1) ) {
-            thishtml += "</div>";
+          $row.append(self.products[i].render());
```

Comment:
`Product.render()` creates DOM now, therefore we refactor the `App.render()` to append DOMs instead of aggregating htmls.